docs/hugo/content/design/ADR-2023-04-Deprecation.md

```diff
@@ -0,0 +1,48 @@
+---
+title: '2023-04: Resource Deprecation'
+---
+
+## Context
+
+As succinctly raised by `@sakthi-vetrivel` in (#2070)[https://github.com/Azure/azure-service-operator/issues/2070]:
+
+> Today, there's no change to the support for an Azure Service Operator CRD when the underlying Azure service/offering is being deprecated. How should we communicate these deprecations and what timelines should we follow? How will we keep our catalog up to date?
+
+Starting in Kubernetes v1.19, it's possible to flag individual versions of a custom resource as (deprecated)[https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-deprecation]. This results in the follow behaviour:
```

Comment:
I believe there is also a CRD max-size, so we may end up with an issue where there are so many supported versions of a resource that we can't fit them all.

#2687 is tracking this.

Unsure if you want to talk about that here at all or not, but in theory we can use a similar approach where we realize we're getting close to the limit and start deprecating older versions.

Reply:
You're absolutely correct, but this ADR is specifically about deprecation of Azure resources, so I don't think that information belongs here.